### PR TITLE
Add cgo worker pool for querynode

### PR DESCRIPTION
**Pull request by congqixia**
Related to #18454 

/kind improvement

Signed-off-by: Congqi Xia <congqi.xia@zilliz.com>

### CODE REVIEW

File: internal/querynode/query_node.go [hunk 1]
```diff
@@ -226,13 +231,22 @@ func (node *QueryNode) Init() error {
 		node.etcdKV = etcdkv.NewEtcdKV(node.etcdCli, Params.EtcdCfg.MetaRootPath)
 		log.Info("queryNode try to connect etcd success", zap.Any("MetaRootPath", Params.EtcdCfg.MetaRootPath))
 
-		node.metaReplica = newCollectionReplica()
+		cpuNum := runtime.GOMAXPROCS(0)
+		node.cgoPool, err = concurrency.NewPool(cpuNum)
```
Comment (yah01) [inline internal/querynode/query_node.go:235]: this pool may be created with ants.WithPreAlloc(true)?

Comment (congqixia) [reply internal/querynode/query_node.go:235]: Good suggestion!

**Pull request merged**